xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -337,28 +337,34 @@ void CGUIDialogAddonInfo::OnInstall()
   if (!m_item->HasAddonInfo())
     return;
 
-  if (m_localAddon)
-  {
-    const std::string& header = g_localizeStrings.Get(19098); // Warning!
-    const std::string text =
-        StringUtils::Format(g_localizeStrings.Get(39028), m_localAddon->ID(),
-                            m_localAddon->Origin(), m_localAddon->Version().asString());
+  const auto& itemAddonInfo = m_item->GetAddonInfo();
+  const std::string& origin = itemAddonInfo->Origin();
 
-    if (CGUIDialogYesNo::ShowAndGetInput(header, text))
-    {
-      m_silentUninstall = true;
-      OnUninstall();
-    }
-    else
+  const AddonRepoUpdateMode updateMode =
+      CAddonSystemSettings::GetInstance().GetAddonRepoUpdateMode();
```

Comment:
No, move this line into the if block.

Or, even better, get rid of the `updateMode` variable:

```c++
if (m_localAddon && (m_localAddon->Origin() != origin) &&
    CAddonSystemSettings::GetInstance().GetAddonRepoUpdateMode() != AddonRepoUpdateMode::ANY_REPOSITORY)
{
  const std::string& header = ...
}
```